prism/src/main/java/org/apache/falcon/service/FeedSLAAlertService.java

```diff
@@ -0,0 +1,173 @@
+        return SERVICE;
+    }
+
+    private FeedSLAAlertService(){}
+
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public void init() throws FalconException {
+        String listenerClassNames = StartupProperties.get().
+                getProperty("feedAlert.listeners");
+        for (String listenerClassName : listenerClassNames.split(",")) {
+            listenerClassName = listenerClassName.trim();
+            if (listenerClassName.isEmpty()) {
+                continue;
+            }
+            EntitySLAListener listener = ReflectionUtils.getInstanceByClassName(listenerClassName);
+            registerListener(listener);
+        }
+
+        String freq = StartupProperties.get().getProperty("feed.sla.statusCheck.frequency.seconds", "600");
+        int statusCheckFrequencySeconds = Integer.parseInt(freq);
+
+        ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(1);
+        executor.scheduleWithFixedDelay(new Monitor(), 0, statusCheckFrequencySeconds + 10, TimeUnit.SECONDS);
+    }
+
+    public void registerListener(EntitySLAListener listener) {
+        listeners.add(listener);
+    }
+
+    @Override
+    public void destroy() throws FalconException {
+
+    }
+
+
+    private class Monitor implements Runnable {
+
+        @Override
+        public void run() {
+            processSLACandidates();
+        }
+    }
+
+    void processSLACandidates(){
+        Multimap<Pair, Date> feedInstancesToBeMonitored = ArrayListMultimap.create();
+//Get all feeds instances to be monitored
+        List<PendingInstanceBean> pendingInstanceBeanList = store.getAllInstances();
+        if (pendingInstanceBeanList.isEmpty()){
+            return;
+        }
+
+        LOG.debug("In processSLACandidates :" + pendingInstanceBeanList.size());
+        try{
+            for (PendingInstanceBean pendingInstanceBean : pendingInstanceBeanList) {
+                Pair pair = new Pair(pendingInstanceBean.getFeedName(), pendingInstanceBean.getClusterName());
+                feedInstancesToBeMonitored.put(pair, pendingInstanceBean.getNominalTime());
+            }
+
+            for(Map.Entry<Pair, Date> pairDateEntry : feedInstancesToBeMonitored.entries()) {
+
+                Pair pair = pairDateEntry.getKey();
+                Date nominalTime = pairDateEntry.getValue();
+                Feed feed = ConfigurationStore.get().get(EntityType.FEED, (String) pair.first);
+
+                Cluster cluster =  FeedHelper.getCluster(feed, (String)pair.second);
+
+                Set<SchedulableEntityInstance> schedulableEntityInstances= FeedSLAMonitoringService.get().
+                        getFeedSLAMissPendingAlerts(feed.getName(), cluster.getName(), nominalTime, nominalTime);
+                if (schedulableEntityInstances.isEmpty()){
+                    store.deleteFeedAlertInstance(feed.getName(), cluster.getName(), nominalTime);
```

Comment:
Shouldn't the method return here, as no further processing is required when the list is empty